dts/bindings/pwm/nxp,ctimer-pwm.yaml

```diff
@@ -0,0 +1,35 @@
+# (c) Meta Platforms, Inc. and affiliates.
+# SPDX-License-Identifier: Apache-2.0
+
+description: NXP CTimer PWM
+
+compatible: "nxp,ctimer-pwm"
+
+include: [pwm-controller.yaml, pinctrl-device.yaml, base.yaml, "nxp,lpc-ctimer.yaml"]
+
+properties:
+  reg:
+    required: true
+
+  prescaler:
+    type: int
+    default: 1
+    description: prescaling value
+
+  period-channel:
+    type: int
+    required: true
+    description: counter for period of pwm signal. must be different than pulse channel
```

Comment:
software configuration in DT, is there any way to avoid this (I think I understand this solution is being proposed there is no period channel in the set cycle API call parameter list)

Reply:
maybe a solution could be having the driver choose the period channel based on whatever channel is not being used as the pulse channel. it would throw an error if it ran out of channels to use as period channels.

Reply:
this is a good idea thanks for finding this solution